### PR TITLE
Implemented instance method imag

### LINKED ISSUE #18226: imag

### PR DESCRIPTION
**Pull request by moezAhmad**
close #18226

### CODE REVIEW

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @moezAhmad ! Thanks for opening this PR. The ToDo list you have linked in the issue actually contains functions that should be added as instance methods for the JAX frontend in `ivy/functional/frontends/jax/devicearray.py`. You can take a look at other functions inside this file to understand how to implement an instance method. Note that, you need to write the test inside `ivy_tests/test_ivy/test_frontends/test_jax/test_jax_devicearray.py`. Again take a look at other tests in this file incase you're stuck. 

Feel free to request another review when you are done with the changes. Thanks

**Comment by moezAhmad** [comment]
@hello-fri-end Thank you for reviewing my pr. I have added instance method for jax fronted as you requested However, my tests are failing locally, giving me the same error for every test "TypeError: 'ArrayImpl' object is not callable". I would appreciate any help in this regard

**Comment by moezAhmad** [comment]
@hello-fri-end I made a mistake by getting math functions and instance methods confused. There's a math function called jax.numpy.imag that wasn't implemented before, due to which I have created a new issue for this according to the doc instruction linked here #20978. I also want to implement imag instance method. I'm ready to start when you can help me.

**Comment by hello-fri-end** [comment]
> @hello-fri-end I made a mistake by getting math functions and instance methods confused. There's a math function called jax.numpy.imag that wasn't implemented before, due to which I have created a new issue for this according to the doc instruction linked here #18981 . I also want to implement imag instance method. I'm ready to start when you can help me.

Hey @moezAhmad ! Thanks for taking the initiative of adding the `imag` functions to the frontends. Now, you can just merge with the master branch here as `imag` as already merged into master. Let me know if you face any issues while doing so :rocket:

**Comment by moezAhmad** [comment]
@hello-fri-end I have merged master as you requested

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
The tests for `test_jax_devicearray_imag` are failing on the CI, can you please take a look?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by moezAhmad** [comment]
@hello-fri-end 
I'm sorry for the delayed response. I've inspected the tests on my end, but I am unable to pinpoint the issue causing the tests for test_jax_devicearray_imag to fail. The specific error I'm encountering is "TypeError: 'ArrayImpl' object is not callable". At this point, I'm unsure of the best way to resolve this. I would greatly appreciate any assistance you can provide in this matter.

File: ivy/functional/frontends/jax/devicearray.py [hunk 1]
```diff
@@ -137,6 +137,10 @@ def sort(self, axis=-1, order=None):
             axis=axis,
             order=order,
         )
+    
+    def imag(self, /):
+        return jax_frontend.numpy.imag(self._ivy_array)
```
Comment (hello-fri-end) [inline ivy/functional/frontends/jax/devicearray.py:142]:
```suggestion
        return jax_frontend.numpy.imag(self)
```

I think this should fix your test :)

**Review by hello-fri-end — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py [hunk 2]
```diff
@@ -324,6 +324,38 @@ def test_jax_devicearray_conj(
         on_device=on_device,
     )
 
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="jax.numpy.array",
+    method_name="imag",
+    dtype_and_x=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("real_and_complex"),
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py:332]:
```suggestion
        available_dtypes=helpers.get_dtypes("valid),
```

Valid automatically determines the supported data-types for the function.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py [hunk 3]
```diff
@@ -324,6 +324,38 @@ def test_jax_devicearray_conj(
...
+        init_all_as_kwargs_np={
+            "object": x[0],
+        },
+        method_input_dtypes=input_dtype,
+        method_all_as_kwargs_np={},
+        frontend=frontend,
+        frontend_method_data=frontend_method_data,
+        init_flags=init_flags,
+        method_flags=method_flags,
+        on_device=on_device,
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_devicearray.py:355]: `backend_to_test = backend_fw`

Can you add this line to the to the `test_frontend_method`? This has been added in a refactor, the test_frontend_function and the test_frontend_function both has this argument now.

**Review by hello-fri-end — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**